src/server/schedulers/serviceScheduler/syncMatomoAccounts.ts

```diff
@@ -0,0 +1,78 @@
+import _ from "lodash";
+
+import { db } from "@/lib/kysely";
+import { getAllUsersInfo } from "@/lib/kysely/queries/users";
+import { Matomo } from "@/lib/matomo";
+import { memberBaseInfoToModel } from "@/models/mapper";
+import { matomoUserToModel } from "@/models/mapper/matomoMapper";
+import { FakeMatomo } from "@/server/config/matomo.config";
+
+export async function syncMatomoAccounts(matomoClient: Matomo | FakeMatomo) {
+    const dbUsers = (await getAllUsersInfo()).map((user) =>
+        memberBaseInfoToModel(user)
+    );
+    const allWebsites = await matomoClient.getAllSites();
+    const matomoUsers = await Promise.all(
+        (
+            await matomoClient.getAllUsers()
+        ).map(async (matomoUser) => {
+            const userMetadata = await matomoClient.fetchUserAccess(
+                matomoUser.serviceUserId
+            );
+            return matomoUserToModel(
+                matomoUser.user,
+                userMetadata,
+                allWebsites
+            );
+        })
+    );
+    const usersToInsert = matomoUsers.map((matomoUser) => {
```

Comment:
pareil ici, on va faire autant de requetes que de matomoUsers
faire une seule requete (avec `user.primary_email IN (x, y, z)`) ou utiliser `p-all`

Reply:
ici il n'y a pas de requete, dbUser est deja chargé